### PR TITLE
[RISCV][GISel] Implement instruction selection for G_PHI and G_BRCOND.

### PR DESCRIPTION
**Pull request by topperc**
This uses a naive lowering for G_BRCOND to a BNE instruction comparing the register to X0.

### CODE REVIEW

File: llvm/test/CodeGen/RISCV/GlobalISel/instruction-select/phi-rv32.mir [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  ; RV32I-NEXT:   $x10 = COPY [[PHI]]
+  ; RV32I-NEXT:   PseudoRET implicit $x10
+  bb.0:
+    liveins: $x10, $x11, $x12
+
+    %0:gprb(s32) = COPY $x10
+    %1:gprb(s32) = COPY $x11
+    %2:gprb(s32) = COPY $x12
+    %3:gprb(s32) = G_CONSTANT i32 1
+    %4:gprb(s32) = G_AND %0, %3
```
Comment (aemerson) [inline llvm/test/CodeGen/RISCV/GlobalISel/instruction-select/phi-rv32.mir:35]: Do you actually need these?

Comment (topperc) [reply llvm/test/CodeGen/RISCV/GlobalISel/instruction-select/phi-rv32.mir:35]: No. I just ran instruction selection on the output of the regbankselect test for G_PHI which also had them. We don't have a post legalizer combiner enabled right now. Should that remove them through computeKnownBits?

File: llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp [hunk 2]
```diff
@@ -205,6 +231,15 @@ bool RISCVInstructionSelector::select(MachineInstr &MI) {
     if (!selectConstant(MI, MIB, MRI))
       return false;
     break;
+  case TargetOpcode::G_BRCOND: {
+    // TODO: Fold with G_ICMP.
+    auto Bcc = MIB.buildInstr(RISCV::BNE)
+      .addReg(MI.getOperand(0).getReg())
+      .addReg(RISCV::X0)
+      .addMBB(MI.getOperand(1).getMBB());
```
Comment (aemerson) [inline llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp:239]: I think you can do something like this:
```
  auto Bcc =
      MIB.buildInstr(RISCV::BNE, {}, {MI.getOperand(0), Register(RISCV::X0)})
          .addMBB(MI.getOperand(1).getMBB());
```
MachineIRBuilder's `SrcOp` can interpret MachineOperands as register implicitly. You can also pass it `RegisterClass`es if you need to implicitly create a vreg.

**Review by aemerson — APPROVED** [review]

**Comment by topperc** [comment]
Commited 98eb28b621d6ea05269c779ceefe0fb7e4862dff

**Pull request closed without merging**